Do not install the python DSDB modules when ADDC is disabled.
This is a change integrated upstream in 4.9.1 (or earlier).

Without this, the python dsdb_dns.so module is installed, which
is linked to libdnsserver-common.so that is NOT installed, and so
potential breakage ensues.

--- a/source4/dsdb/wscript_build	2018-01-17 03:08:39.000000000 -0500
+++ b/source4/dsdb/wscript_build	2018-11-08 10:43:58.064486832 -0500
@@ -63,10 +63,11 @@
 	)
 
 bld.SAMBA_PYTHON('python_dsdb',
 	source='pydsdb.c',
 	# the dependency on dcerpc here is because gensec
 	# depends on dcerpc but the waf circular dependency finder
 	# removes it so we end up with unresolved symbols.
 	deps='samdb pyldb-util dcerpc com_err pyrpc_util pyparam_util dsdb_garbage_collect_tombstones',
-	realname='samba/dsdb.so'
+	realname='samba/dsdb.so',
+	enabled=bld.AD_DC_BUILD_IS_ENABLED()
 	)
--- a/source4/dns_server/wscript_build	2018-11-08 11:49:03.450296286 -0500
+++ b/source4/dns_server/wscript_build	2018-11-08 11:49:23.327864408 -0500
@@ -64,9 +64,10 @@
                   private_library=True,
                   deps='samba-hostconfig samdb-common gensec popt dnsserver_common',
                   enabled=bld.AD_DC_BUILD_IS_ENABLED())
 
 
 bld.SAMBA_PYTHON('python_dsdb_dns',
 	         source='pydns.c',
 	         deps='samdb pyldb-util pyrpc_util dnsserver_common pytalloc-util',
-	         realname='samba/dsdb_dns.so')
+	         realname='samba/dsdb_dns.so',
+	         enabled=bld.AD_DC_BUILD_IS_ENABLED())
